### PR TITLE
Handle root page back press

### PR DESCRIPTION
**Pull request by deepueg**

### CODE REVIEW

File: android/lib/src/main/java/com/ern/api/impl/core/LaunchConfig.java [hunk 1]
```diff
@@ -75,6 +75,11 @@
      */
     boolean startRootInOnCreate = true;
 
+    /***
```
Comment (friederbluemle) [inline android/lib/src/main/java/com/ern/api/impl/core/LaunchConfig.java:78]: Minor: Only _two_ asterisks here for Javadoc.

File: android/lib/src/main/java/com/ern/api/impl/core/LaunchConfig.java [hunk 2]
```diff
@@ -75,6 +75,11 @@
...
+     * Set this to to true if the root page up click is handled by React native.
+     */
+    boolean rootBackPressHandledByRN = false;
```
Comment (friederbluemle) [inline android/lib/src/main/java/com/ern/api/impl/core/LaunchConfig.java:81]: `false` is default for boolean, you can remove it. Also, for more consistency with the other members in this class (except `startRootInOnCreate`), as well as with the Android framework, can you rename this new member variable to `mRootBackPressHandledByRN`?

File: android/lib/src/main/java/com/ern/api/impl/navigation/MenuUtil.java [hunk 3]
```diff
@@ -117,7 +117,7 @@ private static MenuItem addButtonAsMenuItem(@NonNull NavigationBarButton button,
     }
 
     public static boolean canLoadIconFromURI(String icon) {
-        return ElectrodeReactContainer.isReactNativeDeveloperSupport() && URLUtil.isValidUrl(icon) && Patterns.WEB_URL.matcher(icon).matches();
+        return ElectrodeReactContainer.isReactNativeDeveloperSupport() && URLUtil.isValidUrl(icon);
```
Comment (friederbluemle) [inline android/lib/src/main/java/com/ern/api/impl/navigation/MenuUtil.java:120]: Just curious: Why is this no longer needed?

Comment (friederbluemle) [reply android/lib/src/main/java/com/ern/api/impl/navigation/MenuUtil.java:120]: Okay, saw the second commit message. 👍 Maybe we should add some tests?

Comment (deepueg) [reply android/lib/src/main/java/com/ern/api/impl/navigation/MenuUtil.java:120]: This was returning false for a valid URL. Also, we are checking invalid URL before this and that is good enough for the icon check. If the URL has some issues, we catch the exception anyway and move forward safely.  Many devs have raised concerns with the icons not showing up while connected to the local package manager, and this was the root cause.

**Comment by deepueg** [comment]
@friederbluemle Adrressed comments [here](https://github.com/electrode-io/ernnavigation-api-impl-native/pull/107)

**Comment by friederbluemle** [comment]
Thanks @deepueg

**Pull request merged**